domain/src/main/kotlin/no/nav/su/se/bakover/domain/revurdering/VurderOmBeløpsendringErStørreEnnEllerLik10ProsentAvGjeldendeUtbetaling.kt

```diff
@@ -0,0 +1,55 @@
+package no.nav.su.se.bakover.domain.revurdering
+
+import no.nav.su.se.bakover.domain.beregning.Beregning
+import no.nav.su.se.bakover.domain.oppdrag.Utbetalingslinje
+import no.nav.su.se.bakover.domain.tidslinje.Tidslinje
+import kotlin.math.abs
+
+data class VurderOmBeløpsendringErStørreEnnEllerLik10ProsentAvGjeldendeUtbetaling(
+    private val eksisterendeUtbetalinger: List<Utbetalingslinje>,
+    private val nyBeregning: Beregning,
+) {
+    val resultat: Boolean
+
+    init {
+        require(nyBeregning.getMånedsberegninger().none { it.getSumYtelse() == 0 }) { "Kan ikke vurdere endring på 10% for beregninger med beløp lik 0" }
+
+        fun diffEr10ProsentEllerMer(førsteMånedsbeløp: Int, gjeldendeUtbetalingsbeløp: Int) = abs(førsteMånedsbeløp - gjeldendeUtbetalingsbeløp) >= (0.1 * gjeldendeUtbetalingsbeløp)
+
+        val utbetalingstidslinje = Tidslinje(
+            periode = nyBeregning.periode,
+            objekter = eksisterendeUtbetalinger,
+        )
+
+        val førsteMånedsberegning = nyBeregning.getMånedsberegninger()
+            .minByOrNull { it.periode.fraOgMed }!!
+        val gjeldendeUtbetaling = utbetalingstidslinje.gjeldendeForDato(førsteMånedsberegning.periode.fraOgMed)
+
+        resultat = when (gjeldendeUtbetaling) {
+            null -> true
+            is Utbetalingslinje.Ny -> diffEr10ProsentEllerMer(førsteMånedsberegning.getSumYtelse(), gjeldendeUtbetaling.beløp)
+            is Utbetalingslinje.Endring -> {
+                when (gjeldendeUtbetaling.statusendring.status) {
+                    Utbetalingslinje.LinjeStatus.OPPHØR -> {
+                        val opphørsdato = gjeldendeUtbetaling.statusendring.fraOgMed
+                        val opphørGjelderForHeleBeregningsperioden = nyBeregning.getMånedsberegninger()
+                            .map { it.periode }
+                            .map { utbetalingstidslinje.gjeldendeForDato(it.fraOgMed) }
+                            .all { it == gjeldendeUtbetaling }
+                        when {
+                            /**
+                             * Overser 10%-sjekk dersom opphøret gjelder for hele beregningsperioden og første måned
+                             * i beregningen faller på opphørsdato eller senere. I praksis betyr dette at beløpet som
+                             * utbetales er lik 0.
+                             */
+                            opphørGjelderForHeleBeregningsperioden && (opphørsdato.isEqual(førsteMånedsberegning.periode.fraOgMed) || opphørsdato.isBefore(førsteMånedsberegning.periode.fraOgMed)) -> {
+                                diffEr10ProsentEllerMer(førsteMånedsberegning.getSumYtelse(), 0)
+                            }
+                            else -> diffEr10ProsentEllerMer(førsteMånedsberegning.getSumYtelse(), gjeldendeUtbetaling.beløp)
```

Comment:
diffEr10ProsentEllerMer(0,0) == true. Vil det si at vi kan opphøre opphør?

Reply:
I teorien; ja, men i praksis vil flyten vår fange opp "opphør" før vi kommer til 10% sjekken. La vel også inn en sperre på at klassen ikke godtar at noen måneder har beløp med 0.

Ved litt nærmere ettertanke tror jeg kanskje vi må tenke litt nærmere på et par ting før vi tar denne videre:

- Burde vi egentlig også sjekke 10% for tilfeller som i ender opp som opphør?
- Sperren på at 10% sjekken ikke godtar måneder med beløp = 0 vil vel fort fungere dårlig med tilfeller hvor vi revurderer beløp tilbake i tid til å være 0 (vil ikke evalueres til å være et opphør, og vil dermed havne inn i 10% sjekken)